src/mongo/delegates/Views.class.php

```diff
@@ -217,25 +229,22 @@ public function getViewForResources(Array $resources,$viewType,$context=null)
         $missingSubjects = array_diff($resources,$returnedSubjects);
         if (!empty($missingSubjects))
         {
+            $cbdCollection = $this->getCBDCollectionForViewType($viewType);
+            $contextAlias = $this->getContextAlias($context);
+
             $regrabResources = array();
             foreach($missingSubjects as $missingSubject)
             {
-                $viewSpec = $this->getConfigInstance()->getViewSpecification($this->storeName, $viewType);
```

Comment:
I doubt this code is expensive, but it is happening on every iteration of the loop.  I pulled it outside the loop for efficiency (and because I wanted to rationalise the code for reuse in other view-regeneration methods).